Duplicati/Library/Backend/Telegram/TelegramBackend.cs

```diff
@@ -0,0 +1,512 @@
+        {
+            if (options.TryGetValue(Strings.API_ID_KEY, out var apiId))
+            {
+                m_apiId = int.Parse(apiId);
+            }
+
+            if (options.TryGetValue(Strings.API_HASH_KEY, out var apiHash))
+            {
+                m_apiHash = apiHash.Trim();
+            }
+
+            if (options.TryGetValue(Strings.PHONE_NUMBER_KEY, out var phoneNumber))
+            {
+                m_phoneNumber = phoneNumber.Trim();
+            }
+
+            if (options.TryGetValue(Strings.AUTH_CODE_KEY, out var authCode))
+            {
+                m_authCode = authCode.Trim();
+            }
+
+            if (options.TryGetValue(Strings.AUTH_PASSWORD, out var password))
+            {
+                m_password = password.Trim();
+            }
+
+            if (options.TryGetValue(Strings.CHANNEL_NAME, out var channelName))
+            {
+                m_channelName = channelName.Trim();
+            }
+
+            if (m_apiId == 0)
+            {
+                throw new UserInformationException(Strings.NoApiIdError, nameof(Strings.NoApiIdError));
+            }
+
+            if (string.IsNullOrEmpty(m_apiHash))
+            {
+                throw new UserInformationException(Strings.NoApiHashError, nameof(Strings.NoApiHashError));
+            }
+
+            if (string.IsNullOrEmpty(m_phoneNumber))
+            {
+                throw new UserInformationException(Strings.NoPhoneNumberError, nameof(Strings.NoPhoneNumberError));
+            }
+
+            if (string.IsNullOrEmpty(m_channelName))
+            {
+                throw new UserInformationException(Strings.NoChannelNameError, nameof(Strings.NoChannelNameError));
+            }
+
+            m_encSessionStore = new EncryptedFileSessionStore($"{m_apiHash}_{m_apiId}");
+            InitializeTelegramClient();
+        }
+
+        private void InitializeTelegramClient()
+        {
+            m_telegramClient = m_telegramClient ?? new TelegramClient(m_apiId, m_apiHash, m_encSessionStore, m_phoneNumber);
+        }
+
+        public void Dispose()
+        {
+            // Do not dispose m_telegramClient.
+            // There are bugs connected with reusing
+            // the old sockets
+        }
+
+        public string DisplayName { get; } = Strings.DisplayName;
+        public string ProtocolKey { get; } = "telegram";
+
+        public IEnumerable<IFileEntry> List()
+        {
+            return SafeExecute<IEnumerable<IFileEntry>>(() =>
+                {
+                    Authenticate();
+                    EnsureChannelCreated();
+                    var fileInfos = ListChannelFileInfos();
+                    var result = fileInfos.Select(fi => fi.ToFileEntry());
+                    return result;
+                },
+                nameof(List));
+        }
+
+        public Task PutAsync(string remotename, Stream stream, CancellationToken cancelToken)
+        {
+            Test();
```

Comment:
What's the purpose of calling `Test()` here?

Reply:
Ensuring that the client is Authenticated was mandatory. So I replaced Test with Authenticate()